crates/storage/src/bloom.rs

```diff
@@ -121,6 +350,24 @@ impl BloomFilter {
 
         self.check_keys(&filter.keys)
     }
+
+    // Workaround to get the indices of the keys in the filter.
+    // Needed because the `bloomfilter` crate doesn't provide a
+    // way to get this information.
+    fn indices_for_key(key: &Felt) -> Vec<usize> {
+        // Use key on an empty Bloom filter
+        let mut bloom = Self::new();
+        bloom.set(key);
+
+        bloom
+            .0
+            .bit_vec()
+            .iter()
+            .enumerate()
+            .filter(|(_, bit)| *bit)
+            .map(|(i, _)| i)
+            .collect()
+    }
```

Comment:
I think we should just re-implement the pieces of the `bloomfilter` crate we need. We only use a small subset of the features and the API that crate exposes is suboptimal for our use-case anyway.